### PR TITLE
KAFKA-7394; OffsetsForLeaderEpoch supports topic describe access

### PR DESCRIPTION
**Pull request by hachikuji**
As part of KIP-320, allow OffsetsForLeaderEpoch requests with Topic Describe permission.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by hachikuji** [comment]
Note the failing test is fixed in #5635.

File: core/src/main/scala/kafka/cluster/Replica.scala [hunk 1]
```diff
@@ -194,15 +194,16 @@ class Replica(val brokerId: Int,
...
-    replicaString.append("ReplicaId: " + brokerId)
-    replicaString.append("; Topic: " + topicPartition.topic)
-    replicaString.append("; Partition: " + topicPartition.partition)
-    replicaString.append("; isLocal: " + isLocal)
-    replicaString.append("; lastCaughtUpTimeMs: " + lastCaughtUpTimeMs)
+    replicaString.append("Replica(replicaId=" + brokerId)
+    replicaString.append(", topic=" + topicPartition.topic)
+    replicaString.append(", partition=" + topicPartition.partition)
+    replicaString.append(", isLocal=" + isLocal)
+    replicaString.append(", lastCaughtUpTimeMs=" + lastCaughtUpTimeMs)
```
Comment (rajinisivaram) [inline core/src/main/scala/kafka/cluster/Replica.scala]: Perhspa use string interpolation here?

**Review by rajinisivaram — APPROVED** [review]
@hachikuji Thanks for the PR. Left a minor comment, apart from that LGTM.

**Comment by hachikuji** [comment]
I am going to go ahead and merge. The build failures are due to a test regression in streams scala.

**Pull request merged**